integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/tmp/ElasticsearchMetricAggregationsIT.java

```diff
@@ -0,0 +1,165 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hibernate.search.engine.backend.common.DocumentReference;
+import org.hibernate.search.engine.backend.document.IndexFieldReference;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaElement;
+import org.hibernate.search.engine.backend.types.Aggregable;
+import org.hibernate.search.engine.search.aggregation.AggregationKey;
+import org.hibernate.search.engine.search.query.SearchQuery;
+import org.hibernate.search.engine.search.query.SearchResult;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.extension.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.BulkIndexer;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.SimpleMappedIndex;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.StubMappingScope;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class ElasticsearchMetricAggregationsIT {
+
+	@RegisterExtension
+	public static final SearchSetupHelper setupHelper = SearchSetupHelper.create();
+
+	private final SimpleMappedIndex<IndexBinding> mainIndex = SimpleMappedIndex.of( IndexBinding::new ).name( "main" );
+	private final AggregationKey<Integer> sumIntegers = AggregationKey.of( "sumIntegers" );
+	private final AggregationKey<String> sumConverted = AggregationKey.of( "sumConverted" );
+	private final AggregationKey<Integer> minIntegers = AggregationKey.of( "minIntegers" );
+	private final AggregationKey<String> minConverted = AggregationKey.of( "minConverted" );
+	private final AggregationKey<Integer> maxIntegers = AggregationKey.of( "maxIntegers" );
+	private final AggregationKey<String> maxConverted = AggregationKey.of( "maxConverted" );
+	private final AggregationKey<Long> countIntegers = AggregationKey.of( "countIntegers" );
+	private final AggregationKey<Long> countConverted = AggregationKey.of( "countConverted" );
+	private final AggregationKey<Long> countDistinctIntegers = AggregationKey.of( "countDistinctIntegers" );
+	private final AggregationKey<Long> countDistinctConverted = AggregationKey.of( "countDistinctConverted" );
+	private final AggregationKey<Double> avgIntegers = AggregationKey.of( "avgIntegers" );
+	private final AggregationKey<Double> avgConverted = AggregationKey.of( "avgConverted" );
+
+	@BeforeEach
+	void setup() {
+		setupHelper.start().withIndexes( mainIndex ).setup().integration();
+		initData();
+	}
+
+	@Test
+	public void test_filteringResults() {
+		StubMappingScope scope = mainIndex.createScope();
+		SearchQuery<DocumentReference> query = scope.query()
+				.where( f -> f.match().field( "style" ).matching( "bla" ) )
+				.aggregation( sumIntegers, f -> f.sum().field( "integer", Integer.class ) )
+				.aggregation( sumConverted, f -> f.sum().field( "converted", String.class ) )
```

Comment:
Since you have these new metric aggregations extend `AggregationFilterStep` maybe let's  have a test with nested objects and a filter to make sure that there are no surprises there 🙈 😃.

Also, maybe let's try to have some explicit `ValueConvert.YES/NO` ?